numpy/ma/extras.py

```diff
@@ -736,7 +736,7 @@ def _median(a, axis=None, out=None, overwrite_input=False):
     # duplicate high if odd number of elements so mean does nothing
     odd = counts % 2 == 1
     if asorted.ndim > 1:
-        np.copyto(low, high, where=odd)
+        low[odd] = high[odd]
```

Comment:
So now that we are using the `count` function, that may be fixed. In fact, by the time we get there 0-d inputs should have been raveled to 1-d and 1-d arrays have been handled, so the `asorted.ndim > 1` check should not be needed either.

I don't think any of the values should be masked except for zero values of `count`, which brings up the question of negative indices in creating low.